### PR TITLE
Added --log-level option to all components

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/609/)

File: libutils/logging.c [hunk 1]
```diff
@@ -503,6 +503,16 @@ void LogSetGlobalLevel(LogLevel level)
     LoggingPrivSetLevels(level, level);
 }
 
+void LogSetGlobalLevelArgOrExit(const char *const arg)
+{
+    LogLevel level = LogLevelFromString(arg);
+    if (level == LOG_LEVEL_NOTHING)
+    {
+        exit(1);
```
Comment (vpodzime) [inline libutils/logging.c]: Shouldn't this also print an error to STDERR or something?

File: libutils/logging.c [hunk 2]
```diff
@@ -503,6 +503,20 @@ void LogSetGlobalLevel(LogLevel level)
...
 
+void LogSetGlobalLevelArgOrExit(const char *const arg)
+{
+    LogLevel level = LogLevelFromString(arg);
+    if (level == LOG_LEVEL_NOTHING)
+    {
+        // This function is used as part of initializing the logging
+        // system. Using Log() can be considered incorrect, even though
+        // it may "work". Let's just print an error to stderr:
+        fprintf(stderr, "Invalid log level: '%s'\n", arg);
```
Comment (vpodzime) [inline libutils/logging.c]: :+1:

**Comment by vpodzime** [comment]
Btw, I forgot to mention that I really like this PR. How it builds up from the dark underground of our codebase, sheds some light on it and then adds small, isolated improvements. The way to go!

**Pull request merged**